src/resource/bindless.rs

```diff
@@ -0,0 +1,153 @@
+use std::sync::{Arc, Mutex};
+
+use anyhow::Result;
+
+use ash::vk;
+
+use crate::{pipeline::set_layout::DescriptorSetLayoutCreateInfo, util::cache::Resource};
+
+static MAX_BINDLESS_COUNT: u32 = 4096;
+
+pub trait Bindlessable {
```

Comment:
I don't really like this name, it sounds a bit weird. That said I don't immediately know an alternative

Reply:
Mmm yeah it was really hard to find a name for it. What do you think of renaming `BindlessResource` as `BindlessHandle` and this trait as  `BindlessResource`?

Reply:
I think that could work yeah